examples/stateful-precompile/src/main.rs

```diff
@@ -164,15 +164,35 @@ impl ConfigureEvmEnv for MyEvmConfig {
     fn fill_cfg_env(
         &self,
         cfg_env: &mut CfgEnvWithHandlerCfg,
-        header: &Header,
+        header: &Self::Header,
         total_difficulty: U256,
     ) {
         self.inner.fill_cfg_env(cfg_env, header, total_difficulty)
     }
 
+    fn fill_block_env(&self, block_env: &mut BlockEnv, header: &Self::Header, after_merge: bool) {
+        block_env.number = U256::from(header.number);
+        block_env.coinbase = header.beneficiary;
+        block_env.timestamp = U256::from(header.timestamp);
+        if after_merge {
+            block_env.prevrandao = Some(header.mix_hash);
+            block_env.difficulty = U256::ZERO;
+        } else {
+            block_env.difficulty = header.difficulty;
+            block_env.prevrandao = None;
+        }
+        block_env.basefee = U256::from(header.base_fee_per_gas.unwrap_or_default());
+        block_env.gas_limit = U256::from(header.gas_limit);
+
+        // EIP-4844 excess blob gas of this block, introduced in Cancun
+        if let Some(excess_blob_gas) = header.excess_blob_gas {
+            block_env.set_blob_excess_gas_and_price(excess_blob_gas);
+        }
+    }
+
```

Comment:
this code is repeated 3 times. pls open issue to make this the default impl for the trait method when https://github.com/alloy-rs/alloy/issues/1301 is complete.

Reply:
can you elaborate, not really following how that issue is related here

Reply:
Done https://github.com/paradigmxyz/reth/issues/10971

Reply:
@mattsse I have the impression that what is pointed out here is the fact that since we are now using an associated type for header, in the current state of the code, we are no longer able to provide a default implementation for `fill_block_env` in the definition of `ConfigureEvmEnv` directly (because we no longer have access to `header.number` for example).

So we have to repeat this same implementation each time we implement the `ConfigureEvmEnv` trait, which gives us the same piece of code 3 times here.

In alloy it is proposed to add a `BlockHeader` trait that `Header` will have to implement and which will allow direct access to the fields via getters to put this default implementation back in `ConfigureEvmEnv`.

Here is the related PR on alloy as a reference https://github.com/alloy-rs/alloy/pull/1302